### PR TITLE
Wired upsertSearchAttributes to the WorkflowStateMachines

### PR DESCRIPTION
**Pull request by mfateev**
Also changed state machines to not retain passed arguments after a command is created. It will reduce memory consumption.

### CODE REVIEW

File: src/main/java/io/temporal/internal/statemachines/ActivityStateMachine.java [hunk 1]
```diff
@@ -415,5 +404,6 @@ private void createRequestCancelActivityTaskCommand() {
                 RequestCancelActivityTaskCommandAttributes.newBuilder()
                     .setScheduledEventId(getInitialCommandEventId()))
             .build());
+    parameters = null; // avoid retaining large input for the duration of the activity
```
Comment (mastermanu) [inline src/main/java/io/temporal/internal/statemachines/ActivityStateMachine.java:407]: curious what inspired you to make this change (whether it just occurred to you or you saw something interesting during local debugging / execution)

File: src/test/java/io/temporal/internal/statemachines/MutableSideEffectStateMachineTest.java [hunk 2]
```diff
@@ -34,15 +38,46 @@
...
+        MutableSideEffectStateMachine.STATE_MACHINE_DEFINITION.getUnvisitedTransitions(
+            stateMachineList);
+    if (!missed.isEmpty()) {
+      CommandsGeneratePlantUMLStateDiagrams.writeToFile(
+          "test",
+          MutableSideEffectStateMachine.class,
+          MutableSideEffectStateMachine.STATE_MACHINE_DEFINITION.asPlantUMLStateDiagramCoverage(
+              stateMachineList));
+      fail(
+          "SideEffectStateMachine is missing test coverage for the following transitions:\n"
```
Comment (mastermanu) [inline src/test/java/io/temporal/internal/statemachines/MutableSideEffectStateMachineTest.java:76]: [Nit] Fix error message to say MutableSideEffectStateMachine

File: src/test/java/io/temporal/internal/statemachines/TimerStateMachineTest.java [hunk 3]
```diff
@@ -34,15 +35,42 @@
...
+    List<Transition> missed =
+        TimerStateMachine.STATE_MACHINE_DEFINITION.getUnvisitedTransitions(stateMachineList);
+    if (!missed.isEmpty()) {
+      CommandsGeneratePlantUMLStateDiagrams.writeToFile(
+          "test",
+          TimerStateMachine.class,
+          TimerStateMachine.STATE_MACHINE_DEFINITION.asPlantUMLStateDiagramCoverage(
+              stateMachineList));
+      fail(
+          "SideEffectStateMachine is missing test coverage for the following transitions:\n"
```
Comment (mastermanu) [inline src/test/java/io/temporal/internal/statemachines/TimerStateMachineTest.java:69]: [Nit] fix error message to say TimerStateMachine

File: src/main/java/io/temporal/internal/statemachines/ActivityStateMachine.java (hunk 1 above)
Comment (mfateev) [reply src/main/java/io/temporal/internal/statemachines/ActivityStateMachine.java:407]: I was thinking about it in my spare time but was forgetting about it when getting to the computer :).

**Pull request merged**